src/main/java/com/synopsys/integration/alert/channel/jira/server/JiraServerIssueConfigValidator.java

```diff
@@ -138,14 +141,23 @@ private String validateIssueCreator(FieldAccessor fieldAccessor, Map<String, Str
     }
 
     private String validateIssueType(FieldAccessor fieldAccessor, Map<String, String> fieldErrors) {
+        Optional<String> optionalProjectName = fieldAccessor.getString(JiraServerDescriptor.KEY_JIRA_PROJECT_NAME);
         String issueType = fieldAccessor.getString(JiraServerDescriptor.KEY_ISSUE_TYPE).orElse(JiraServerDistributionUIConfig.DEFAULT_ISSUE_TYPE);
         try {
             boolean isValidIssueType = issueTypeService.getAllIssueTypes()
                                            .stream()
                                            .map(IssueTypeResponseModel::getName)
                                            .anyMatch(issueType::equals);
             if (isValidIssueType) {
-                return issueType;
+                if (optionalProjectName.isPresent()) {
+                    String projectName = optionalProjectName.get();
+                    boolean isValidForProject = issueMetaDataService.doesProjectContainIssueType(projectName, issueType);
+                    if (isValidForProject) {
+                        return issueType;
+                    } else {
+                        fieldErrors.put(JiraServerDescriptor.KEY_ISSUE_TYPE, String.format("The issue type '%s' not assigned to project '%s'", issueType, projectName));
+                    }
+                }
```

Comment:
Looks like we're starting to duplicate code between the Jira instances. Would be nice if we could split this out somewhere.